src/drawGenerator.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+  require_once "src/abstract.php";
+  require_once "src/interface.php";
+
+class Square extends Shape implements IShapeDrawSquare, IShapeSetColor
+{
+  protected $length = 0;
+
+  private $colors = array("red" => 0, "green" => 0, "blue" => 0);
+
+  private $isRandom = false;
+
+  public function show() {
+    if(!($this->checkSidesValue($this->length))) 
+      return false;
+
+    $this->draw();
+  }
+  
+  public function getShapeName() : string {
+    return $this->shapeName;
+  }
+
+  public function draw() {
+    for($i = 0; $i < $this->length; ++$i) {
+      for($k = 0; $k < $this->length; ++$k) {
+        echo "<div class='block' style='float: left;
+        height: ".Shape::DRAWING_SCALE_X."px; width: ".Shape::DRAWING_SCALE_Y."px; background-color: ".$this->getRGBtoHex().";'></div>";
+      }
+      echo "<div style='clear:both;'></div><br>";
+    }
+  }
+
+  public function checkSidesValue($length) {
+    if($length > 0)
+      return true;
+    else
+      return false;
+  }
+
+  public function getValueSide() {
+    return array("length" => $this->length);
+  }
+
+  public function setValueSide($length) {
+    if($this->checkSidesValue($length)) {
+      $this->length = $length;
+      return true;
+    }
+  }
+
+  public function setColorBlocks($color_red, $color_green, $color_blue) {
+    if($color_red > -1 && $color_red < 256 && $color_green > -1 && $color_green < 256 && $color_blue > -1 && $color_blue < 256) {
```

Comment:
You can extract private function isClorValid() and here do like 
If(isColorValid($color_red) && isColorValid($color_green)  && isColorValid($color_blue))

Or even better create class Color that have validation in constructor and ColorType enum that will have values red gren blue

Reply:
I can't create private methods in interfaces :(
I created isColorValid();